fiftyone/utils/transformers.py

```diff
@@ -0,0 +1,270 @@
+                logits=logits[i],
+            )
+            for i in range(logits.shape[0])
+        ]
+
+
+def _convert_bounding_box(box, image_shape):
+    top_left_x, top_left_y, bottom_right_x, bottom_right_y = box
+
+    width = bottom_right_x - top_left_x
+    height = bottom_right_y - top_left_y
+
+    img_width, img_height = image_shape
+
+    return [
+        top_left_x / img_width,
+        top_left_y / img_height,
+        width / img_width,
+        height / img_height,
+    ]
+
+
+def _to_detections(result, id2label, image_size):
+    detections = []
+
+    scores = result["scores"].cpu().numpy()
+    labels = result["labels"].cpu().numpy()
+    boxes = result["boxes"].cpu().numpy()
+    for score, label, box in zip(scores, labels, boxes):
+        box = [round(i, 2) for i in box.tolist()]
+        box = _convert_bounding_box(box, image_size)
+        detections.append(
+            fol.Detection(
+                label=id2label[label.item()],
+                bounding_box=box,
+                confidence=score.item(),
+            )
+        )
+
+    return fol.Detections(detections=detections)
+
+
+def to_detections(results, id2label, image_sizes):
+    if isinstance(results, dict):
+        return _to_detections(results, id2label, image_sizes[0])
+    elif len(results) == 1:
+        return _to_detections(results[0], id2label, image_sizes[0])
+    else:
+        return [
+            _to_detections(result, id2label, image_sizes[i])
+            for i, result in enumerate(results)
+        ]
+
+
+class FiftyOneTransformerConfig(Config):
+    """Configuration for a :class:`FiftyOneTransformer`.
+
+    Args:
+        model (None): a ``transformers.model`` model
+        name_or_path (None): the name or path to a checkpoint file to load
+    """
+
+    def __init__(self, d):
+        self.model = self.parse_raw(d, "model", default=None)
+        self.name_or_path = self.parse_string(d, "name_or_path", default=None)
+
+
+class FiftyOneTransformer(Model):
+    """FiftyOne wrapper around a ``transformers.model`` model.
+
+    Args:
+        config: a `FiftyOneTransformerConfig`
+    """
+
+    def __init__(self, config):
+        self.config = config
+        self.model = self._load_model(config)
+        self.image_processor = self._load_image_processor()
+
+    @property
+    def media_type(self):
+        return "image"
+
+    @property
+    def ragged_batches(self):
+        return False
+
+    @property
+    def transforms(self):
+        return None
+
+    @property
+    def preprocess(self):
+        return False
+
+    def _load_image_processor(self):
+        return _get_image_processor(self.model)
+
+    def _load_model(self, config):
+        raise NotImplementedError("Subclass must implement _load_model()")
+
+    def predict(self, arg):
+        raise NotImplementedError("Subclass must implement predict()")
+
+
+class FiftyOneTransformerForImageClassification(FiftyOneTransformer):
```

Comment:
@jacobmarks think you could implement the `fiftyone.core.models.EmbeddingsMixin` interface here so that HF classification models would support embeddings?

Here's how that was implemented for Torch models:
https://github.com/voxel51/fiftyone/blob/d16ed6946b3abaccf0ba1a7c975c25bc1b365df6/fiftyone/utils/torch.py#L247

Reply:
Absolutely! Can do